### PR TITLE
Fix: Mini-Cart block shows wrong total if theres multiple installs on the same domain

### LINKED ISSUE #11168: Mini-Cart block shows wrong total if theres multiple installs on the same domain
## Describe the bug

If you have multiple WooCommerce sites on the same domain, the mini-cart block will often show the wrong cart total in case you started a cart on another site. 

## To reproduce

1. Setup _Site A_ on domain.com, lets use EUR as the currency
2. Setup _Site B_ on domain.com/test and use USD as the currency
3. Add something to the cart on _Site A_
4. Go to _Site B_ and the mini-cart will show the cart total coming from _Site A_, even the currency will be EUR. 

## Expected behavior

Obviously it should display the correct cart total in this case. 

## Additional context

I'm guessing this happens because some of the mini-cart data is stored in local storage and that is tied to the domain only, not the path. To give you a real-world use case, some multilingual stores might be setup on /en and /de paths as separate WP/WC installs.

### PR DESCRIPTION
**Pull request by dinhtungdu**
## What

This PR fixes the issue when Mini Cart block, when is used in a sub directory, failed to fetch the cart or fetch the cart of the Woo store under the root domain.

Fixes #11168

## Why

The `getMiniCartTotalsFromServer()` was using a hard-coded URL to fetch the cart data, it works when the Woo store URL is the root domain but when the store uses sub directory, it doesn't. This PR fixes that issue by using `@wordpress/api-fetch` instead of browser's `fetch`.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Setup _Site B_ on domain.com/test and use USD as the currency
5. See the Site B Mini Cart works as expected.
1. Setup _Site A_ on domain.com, lets use EUR as the currency
3. Add something to the cart on _Site A_
4. Go to _Site B_ and **don't see** its mini-cart affected by site A.

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Fix the Mini Cart button shows incorrect data for stores under sub-directories.

### CODE REVIEW

File: assets/js/blocks/mini-cart/utils/data.ts [hunk 1]
```diff
@@ -128,15 +129,9 @@ export const getMiniCartTotalsFromLocalStorage = ():
...
-	return fetch( '/wp-json/wc/store/v1/cart/' )
-		.then( ( response ) => {
-			// Check if the response was successful.
-			if ( ! response.ok ) {
-				throw new Error();
-			}
-
-			return response.json();
-		} )
+	return apiFetch< CartResponse >( {
```
Comment (tarunvijwani) [inline assets/js/blocks/mini-cart/utils/data.ts:132]: @dinhtungdu Thank you for the fix! Just for my learning, how apiFetch is different than browser fetch?

In the [docs](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-api-fetch/), it says `It’s a wrapper around window.fetch.`

Comment (dinhtungdu) [reply assets/js/blocks/mini-cart/utils/data.ts:132]: @tarunvijwani It comes with some middleware, one of them is `createRootURLMiddleware` which WP uses to set the correct root API URL in `wp_default_packages_inline_scripts()`:
```
	$scripts->add_inline_script(
		'wp-api-fetch',
		sprintf(
			'wp.apiFetch.use( wp.apiFetch.createRootURLMiddleware( "%s" ) );',
			sanitize_url( get_rest_url() )
		),
		'after'
	);
```

So by using `apiFetch`, we can always fetch the correct endpoint by just providing the path.

Comment (tarunvijwani) [reply assets/js/blocks/mini-cart/utils/data.ts:132]: That's really helpful. Thank you!

**Review by tarunvijwani — APPROVED** [review]
LGTM! 👍

**Comment by tjcafferkey** [comment]
@dinhtungdu this PR should have fixed this issue, right? https://github.com/woocommerce/woocommerce-blocks/issues/10307
